File: src/main/java/ru/gdlbo/search/searcher/repository/UserRole.java
package ru.gdlbo.search.searcher.repository;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Setter
@Getter
@Entity
@Table(name = "user_roles")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(exclude = "user")
public class UserRole {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String role;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User user;

    public UserRole(String role, User user) {
        this.role = role;
        this.user = user;
    }
}

File: src/main/java/ru/gdlbo/search/searcher/repository/UserRole.kt
package ru.gdlbo.search.searcher.repository

import jakarta.persistence.*
import lombok.*

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "user_roles")
@EqualsAndHashCode(exclude = ["user", "role"])
class UserRole(
    @JoinColumn(name = "user_id", nullable = false)
    @ManyToOne
    private val user: User,

    @JoinColumn(name = "role_id", nullable = false)
    @ManyToOne
    val role: Role
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    constructor() : this(User(), Role())
}